src/compiler/transformers/ts.ts

```diff
@@ -1794,12 +1794,12 @@ namespace ts {
          * @param node The node that should have its return type serialized.
          */
         function serializeReturnTypeOfNode(node: Node): SerializedTypeNode {
-            if (isFunctionLike(node) && node.type) {
-                return serializeTypeNode(node.type);
-            }
-            else if (isAsyncFunction(node)) {
+            if (isAsyncFunction(node)) {

```

Comment:
This isn't correct if you are targeting ES5 with async functions and a third-party `Promise` implementation:

```ts
import { SomePromise } from "some-package";
class C {
  @dec
  async function foo(): SomePromise {
  }
}
```

In that case, we want to emit `SomePromise`, not `Promise`.

You probably need to add an additional check that the async function either:

- Has no return type
- Has a return type that cannot be resolved as a value and we are in `isolatedModules`.